modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/manager/v1_0/DefaultObjectEntryManagerImpl.java

```diff
@@ -1078,6 +1108,33 @@ private void _checkObjectEntryObjectDefinitionId(
 		}
 	}
 
+	private ObjectEntry _copyVersionedObjectEntry(
+			DTOConverterContext dtoConverterContext, ObjectEntry objectEntry,
+			ObjectDefinition objectDefinition)
+		throws Exception {
+
+		Map<String, Object> properties = objectEntry.getProperties();
+
+		List<ObjectField> objectFields =
+			_objectFieldLocalService.getCustomObjectFields(
+				objectDefinition.getObjectDefinitionId());
+
+		for (ObjectField objectField : objectFields) {
+			if (ObjectFieldUtil.isReadOnly(
+					ddmExpressionFactory, objectEntry.getProperties(),
+					objectField)) {
+
+				properties.remove(objectField.getName());
+			}
+		}
+
+		objectEntry.setProperties(() -> properties);
```

Comment:
I think this is redundant since `properties` is just a reference copy of the original map (`objectEntry.getProperties()`).

Can you test it to confirm?

Reply:
Tested and it worked, good catch!